### PR TITLE
Uses server-defined currency string throughout app.

### LINKED ISSUE #1501: Eigen hard codes currency symbol for Auction Bid Amounts
**Issue:** Auctions can now be run in non USD currencies. However Eigen seems to hardcode the currency symbol to "$"

**Details:**
I created a mock auction in staging and changed its currency to "GBP". Force and MicroG bidding worked for the most part without kinks. Eigen showed Esimates in the correct currency but the "Bid Amount" on both the Auction Landing Page and the Artwork page was hardcoded to "$"

image

### PR DESCRIPTION
**Pull request by ashfurrow**
Fixes #1501.

### CODE REVIEW

**Comment by ashfurrow** [comment]
I don't think @mention-bot has ever recommended someone other than Orta as a reviewer.

**Comment by orta** [comment]
👍 Yep - this looks good, off the top of my head I can only thing of one place that it's worth double checking we touch, I think one of the places we show prices is in the `AREmbeddedModels` - though that usually doesn't come from SaleArtwork - but is worth checking just in case

**Comment by ashfurrow** [comment]
Hmm, yeah. `ARArtworkThumbnailMetadataView` accesses the artwork's `price` but is never parsed and also never displayed. Auctions-related embedded model cells use `SaleArtwork`'s `currentOrStartingBidWithNumberOfBids` and `+dollarsFromCents:currencySymbol:`. Just realized the metricification extension isn't using its new `currencySymbol` parameter. Will amend.

**Comment by ArtsyMobile** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-kind="Warning">
        
            2 Warnings
        
      </th>
     </tr>
  </thead>
  <tbody>
        <tr>
      <td>:warning:</td>
      <td data-sticky="true">Needs testing on a Phone if change is non-trivial</td>
    </tr>
        <tr>
      <td>:warning:</td>
      <td data-sticky="true">Detected some Swift building time outliers</td>
    </tr>
    
    
  </tbody>
</table>
  

  



Time | Class | Function |
| --- | ----- | ----- |247.9ms | [AuctionViewController.swift](/artsy/eigen/tree/ashfurrow-currency-handling/Artsy/View_Controllers/Auction/AuctionViewController.swift#L176) | @objc dynamic func showRefineTappedAnimated(animated: Bool)
225.6ms | [SwiftyJSON.swift](/artsy/eigen/tree/ashfurrow-currency-handling/Pods/SwiftyJSON/Source/SwiftyJSON.swift#L445) | get {}
179.8ms | [WebSocket.swift](/artsy/eigen/tree/ashfurrow-currency-handling/Pods/Starscream/Source/WebSocket.swift#L528) | private func processRawMessage(buffer: UnsafePointer<UInt8>, bufferLen: Int)




<p align="right" data-meta="generated_by_danger" data-base-commit="" data-head-commit="" >
  Generated by :no_entry_sign: <a href="https://github.com/danger/danger/">danger</a>
</p>

**Comment by ashfurrow** [comment]
Okay, a few issues resolved. I had the metric extension of `Int` using `SaleArtwork`'s class method for formatting, but I was getting really inconsistent results in testing. _Maybe_ it's a weird interop thing with Swift/ObjC and Quick/Specta, not sure. In any case, it's working, here are some samples:

screen shot 2016-05-16 at 6 48 33 pm
screen shot 2016-05-16 at 6 48 26 pm
screen shot 2016-05-16 at 6 48 21 pm

**Comment by devangt** [comment]
👍

File: Artsy/Categories/Apple/Int+Metrification.swift [hunk 1]
```diff
@@ -1,7 +1,26 @@
 import Foundation
 
-// This module has the NSNumberFormatter methods we need
-import Artsy_UILabels
+private struct Formatter {
+    private static var _formmater: NSNumberFormatter?
```
Comment (orta) [inline Artsy/Categories/Apple/Int+Metrification.swift]: typo ^

File: Artsy/Categories/Apple/Int+Metrification.swift [hunk 2]
```diff
@@ -1,7 +1,26 @@
...
+    private static var formatter: NSNumberFormatter {
+        guard let formatter = _formmater else {
+            let newFormatter = NSNumberFormatter()
+            newFormatter.locale = NSLocale.currentLocale()
+            newFormatter.currencyCode = "USD"
+            newFormatter.numberStyle = .CurrencyStyle
+            newFormatter.maximumFractionDigits = 0
+            newFormatter.alwaysShowsDecimalSeparator = false
+            return newFormatter
+        }
```
Comment (orta) [inline Artsy/Categories/Apple/Int+Metrification.swift]: but nice pattern 👍

**Comment by orta** [comment]
Yep - this is good, fix the typo and this can get merged 👍

**Comment by ashfurrow** [comment]
🍏

**Comment by ashfurrow** [comment]
A wide-reaching PR, only change since given he 👍 is a typo fix, going to merge to prevent conflicts later on.

**Pull request merged**